File: bl-central-server/bl-sensor-data-collector/src/main/java/org/jesperancinha/logistics/sensor/collector/configuration/CollectorConfiguration.java
package org.jesperancinha.logistics.sensor.collector.configuration;

import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
import org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer;
import org.springframework.amqp.rabbit.listener.adapter.MessageListenerAdapter;
import org.springframework.beans.factory.annotation.Value;

public abstract class CollectorConfiguration {

    @Value("${spring.rabbitmq.username}")
    private String username;

    @Value("${spring.rabbitmq.password}")
    private String password;

    @Value("${spring.rabbitmq.host}")
    private String host;

    protected SimpleMessageListenerContainer getSimpleMessageListenerContainer(MessageListenerAdapter listenerAdapter, String vHost, String queueName) {
        SimpleMessageListenerContainer container = new SimpleMessageListenerContainer();
        CachingConnectionFactory connectionFactory = new CachingConnectionFactory(host);
        connectionFactory.setUsername(username);
        connectionFactory.setPassword(password);
        connectionFactory.setVirtualHost(vHost);
        container.setConnectionFactory(connectionFactory);
        container.setQueueNames(queueName);
        container.setMessageListener(listenerAdapter);
        return container;
    }
}

File: bl-central-server/bl-sensor-data-collector/src/main/kotlin/org/jesperancinha/logistics/sensor/collector/configuration/CollectorConfiguration.kt
package org.jesperancinha.logistics.sensor.collector.configuration

import org.springframework.amqp.rabbit.connection.CachingConnectionFactory
import org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer
import org.springframework.amqp.rabbit.listener.adapter.MessageListenerAdapter
import org.springframework.beans.factory.annotation.Value

abstract class CollectorConfiguration {
    @Value("\${spring.rabbitmq.username}")
    private val username: String? = null

    @Value("\${spring.rabbitmq.password}")
    private val password: String? = null

    @Value("\${spring.rabbitmq.host}")
    private val host: String? = null
    protected fun getSimpleMessageListenerContainer(
        listenerAdapter: MessageListenerAdapter?,
        vHost: String?,
        queueName: String?
    ): SimpleMessageListenerContainer {
        val container = SimpleMessageListenerContainer()
        val connectionFactory = CachingConnectionFactory(host)
        connectionFactory.username = username
        connectionFactory.setPassword(password)
        connectionFactory.virtualHost = vHost
        container.connectionFactory = connectionFactory
        container.setQueueNames(queueName)
        container.setMessageListener(listenerAdapter)
        return container
    }
}